#VRML_SIM V6.0 utf8
WorldInfo {
  info [
    "Cozmo test world. All rights reserved. Copyright (c) Anki, Inc."
    "Author: Hanns Tappeiner <hanns@anki.com>"
    "Date: DD MMM YYYY"
    ""
  ]
  basicTimeStep 16
  contactProperties [
    DEF FLOOR_FRONTWHEEL ContactProperties {
      material1 "cmat_floor"
      material2 "highfriction"
      coulombFriction 50
      bounce 0.01
    }
    DEF RAMP_FRONTWHEEL ContactProperties {
      material1 "cmat_ramp"
      material2 "highfriction"
      coulombFriction 50
    }
    DEF BLOCK_FRONTWHEEL ContactProperties {
      material1 "cmat_block"
      material2 "highfriction"
      coulombFriction 50
      bounce 0.01
    }
    DEF BLOCK_REARWHEEL ContactProperties {
      material1 "cmat_block"
      material2 "nofriction"
      coulombFriction 0.0001
      bounce 0.001
    }
    DEF FLOOR_REARWHEEL ContactProperties {
      material1 "cmat_floor"
      material2 "nofriction"
      coulombFriction 0.0001
      bounce 0.001
    }
    DEF RAMP_REARWHEEL ContactProperties {
      material1 "cmat_ramp"
      material2 "nofriction"
      coulombFriction 0.0001
      bounce 0.001
    }
    DEF FLOOR_BLOCK ContactProperties {
      material1 "cmat_floor"
      material2 "cmat_block"
      coulombFriction 5000
      bounce 0.001
    }
    DEF BLOCK_BLOCK ContactProperties {
      material1 "cmat_block"
      material2 "cmat_block"
      coulombFriction 5000
      bounce 0.001
    }
    DEF RAMP_BLOCK ContactProperties {
      material1 "cmat_ramp"
      material2 "cmat_block"
      coulombFriction 5000
      bounce 0.001
    }
    DEF RAMP_RAMP ContactProperties {
      material1 "cmat_ramp"
      material2 "cmat_ramp"
      coulombFriction 5000
      bounce 0.001
    }
    DEF FLOOR_RAMP ContactProperties {
      material1 "cmat_ramp"
      material2 "cmat_block"
      coulombFriction 5000
      bounce 0.001
    }
  ]
}
Viewpoint {
  orientation -0.317568 -0.871481 -0.373726 1.31349
  position -0.772285 0.821524 0.312409
}
Background {
  skyColor [
    0.4 0.7 1
  ]
}
DirectionalLight {
  ambientIntensity 0.5
  direction 1 -2 1
  castShadows TRUE
}
DEF FLOOR Solid {
  translation -0.5 0 -0.5
  children [
    Shape {
      geometry ElevationGrid {
        color Color {
          color [
            0.43653 0.598901 0.288273
            1 1 0
          ]
        }
        colorPerVertex FALSE
        xDimension 11
        xSpacing 0.1
        zDimension 11
        zSpacing 0.1
      }
    }
  ]
  contactMaterial "cmat_floor"
  boundingObject Transform {
    translation 0.5 0 0.5
    children [
      Plane {
      }
    ]
  }
}
Block1x1 {
  translation 0.2 0.09 0.2
}
Ramp2x1 {
  translation 0 0.09 -0.28
}
Ramp2x1 {
  translation 0.3 0.03 -0.18
}
Ramp2x1 {
  translation 0 0.03 -0.059
}
DEF WALL_1 Solid {
  translation 0.5 0.025 0
  children [
    DEF WALL_SHAPE Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.12549 0.368627 0.729412
        }
      }
      geometry Box {
        size 0.02 0.05 0.98
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -0.5 0.025 0
  children [
    USE WALL_SHAPE
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.025 -0.5
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0 0.025 0.5
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  boundingObject USE WALL_SHAPE
}
DEF Cozmo Robot {
  translation 0 0.018 0.2
  rotation 9.30088e-10 -1.52937e-07 -1 7.48661e-07
  children [
    Solid {
      translation 0 0.034 0.02
      children [
        DEF LBase Shape {
          geometry Box {
            size 0.01 0.02 0.01
          }
        }
        Transform {
          children [
            HingeJoint {
              jointParameters HingeJointParameters {
              }
              device RotationalMotor {
                name "lift_motor"
              }
              endPoint Solid {
                translation 0 0 -0.035
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      anchor 0 0 -0.035
                    }
                    device RotationalMotor {
                      name "lift_motor2"
                    }
                    endPoint Solid {
                      translation 0 0 -0.035
                      children [
                        DEF CAM_LIFT Camera {
                          translation 0 0 -0.001
                          name "cam_lift"
                          width 320
                          height 240
                          near 0.001
                          windowPosition 0.592267 0.104631
                          pixelSize 0.5
                        }
                        Connector {
                          translation 0 0 -0.0025
                          rotation 0 1 0 3.14159
                          model "magnetic"
                          type "active"
                          distanceTolerance 0.006
                          snap FALSE
                        }
                        DEF LIFTFRONT Shape {
                          geometry Box {
                            size 0.03 0.01 0.005
                          }
                        }
                      ]
                      boundingObject USE LIFTFRONT
                      physics Physics {
                        density 1
                      }
                    }
                  }
                  DEF LBottom Shape {
                    geometry Box {
                      size 0.001 0.005 0.07
                    }
                  }
                ]
                boundingObject USE LBottom
                physics Physics {
                  density 1
                  centerOfMass [
                    0 0 0
                  ]
                }
              }
            }
          ]
        }
      ]
      boundingObject USE LBase
      physics Physics {
        density 50000
      }
    }
    DEF BODY Shape {
      appearance Appearance {
        material DEF MAT_WHITE Material {
          diffuseColor 1 1 1
        }
      }
      geometry Box {
        size 0.014 0.014 0.048
      }
    }
    Solid {
      translation 0 0.002 -0.015
      children [
        DEF BODYA Shape {
          appearance Appearance {
            material DEF MAT_WHITE Material {
              diffuseColor 1 1 1
            }
          }
          geometry Box {
            size 0.014 0.018 0.02
          }
        }
      ]
      boundingObject USE BODYA
      physics Physics {
        density 5000
      }
    }
    DEF WHEEL_REAR_RIGHT HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.02 0 0.024
      }
      endPoint Solid {
        translation 0.02 3.92458e-06 0.024
        rotation 1 0 0 0
        children [
          DEF WHEEL_BOX Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.505501 0.505501 0.505501
              }
            }
            geometry Box {
              size 0.0204 0.002 0.01
            }
          }
          DEF WHEEL Transform {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.254032 0.254032 0.254032
                  }
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.015
                  subdivision 36
                }
              }
            ]
          }
        ]
        contactMaterial "nofriction"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL_REAR_LEFT HingeJoint {
      jointParameters HingeJointParameters {
        position -1.15762e-11
        anchor -0.02 0 0.024
      }
      endPoint Solid {
        translation -0.02 3.92342e-06 0.024
        rotation 0.0797287 -2.06116e-06 0.996817 -1.15774e-11
        children [
          USE WHEEL_BOX
          USE WHEEL
        ]
        contactMaterial "nofriction"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL_FRONT_RIGHT HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.02 0 -0.024
      }
      device RotationalMotor {
        name "wheel_fr"
      }
      endPoint Solid {
        translation 0.02 3.92458e-06 -0.024
        rotation 1 0 0 0
        children [
          USE WHEEL_BOX
          USE WHEEL
        ]
        contactMaterial "highfriction"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL_FRONT_LEFT HingeJoint {
      jointParameters HingeJointParameters {
        position -1.15794e-11
        anchor -0.02 0 -0.024
      }
      device RotationalMotor {
        name "wheel_fl"
      }
      endPoint Solid {
        translation -0.02 3.92342e-06 -0.024
        rotation 0.0794063 -2.05874e-06 0.996842 -1.15799e-11
        children [
          USE WHEEL_BOX
          USE WHEEL
        ]
        contactMaterial "highfriction"
        boundingObject USE WHEEL
        physics Physics {
          centerOfMass [
            0 0 0
          ]
        }
      }
    }
    DEF HEAD HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0 0.02 0
      }
      device RotationalMotor {
        name "motor_head_pitch"
      }
      endPoint Solid {
        rotation 1 0 0 0
        children [
          DEF CAM_HEAD Camera {
            translation 0 0.028 -0.025
            rotation 1 0 0 0
            children [
              Transform {
                rotation 1 0 0 -0.7
                children [
                  SpotLight {
                    beamWidth 0.25
                    cutOffAngle 0.8
                    direction 0 1 -1
                    intensity 0.8
                  }
                ]
              }
            ]
            name "cam_head"
            width 640
            height 480
            near 0.001
            pixelSize 0.5
            colorNoise 0.03
          }
          DEF EYE_LEFT Transform {
            translation -0.013 0.034 -0.01
            rotation 1 0 0 -1.5708
            children [
              DEF EYE_SHAPE Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 1 1 1
                  }
                }
                geometry Cylinder {
                  height 0.04
                  radius 0.012
                  subdivision 24
                }
              }
            ]
          }
          DEF EYE_RIGHT Transform {
            translation 0.013 0.034 -0.01
            rotation 1 0 0 -1.5708
            children [
              USE EYE_SHAPE
            ]
          }
        ]
      }
    }
  ]
  name "Cozmo"
  boundingObject DEF BODYA Shape {
    appearance Appearance {
      material DEF MAT_WHITE Material {
        diffuseColor 1 1 1
      }
    }
    geometry Box {
      size 0.014 0.018 0.02
    }
  }
  physics Physics {
    density -1
    mass 1
  }
  locked TRUE
  controller "cozmo_basicdrive01"
}
Block1x1 {
  translation 0 0.03 -0.3
}
Block1x1 {
  translation 0 0.03 -0.09
}
Block1x1 {
  translation 0 0.03 -0.151
}
Block1x1 {
  translation -0.061 0.03 -0.151
}
Block1x1 {
  translation -0.122 0.03 -0.151
}
Block1x1 {
  translation -0.183 0.03 -0.151
}
Block1x1 {
  translation -0.183 0.09 -0.151
}
Block1x1 {
  translation 0.2 0.03 -0.1
  rotation 0 1 0 0.3927
}
Block1x1 {
  translation 0.4 0.03 0
  rotation 0 1 0 -0.1309
}
Block1x1 {
  translation 0.2 0.03 0.2
}
